src/model_theory/language_map.lean

```diff
@@ -174,12 +174,22 @@ protected structure injective : Prop :=
 /-- A language homomorphism is an expansion on a structure if it commutes with the interpretation of
 all symbols on that structure. -/
 class is_expansion_on (M : Type*) [L.Structure M] [L'.Structure M] : Prop :=
-(map_on_function : ∀ {n} (f : L.functions n) (x : fin n → M),
+(map_on_function' : ∀ {n} (f : L.functions n) (x : fin n → M),
   fun_map (ϕ.on_function f) x = fun_map f x)
-(map_on_relation : ∀ {n} (R : L.relations n) (x : fin n → M),
+(map_on_relation' : ∀ {n} (R : L.relations n) (x : fin n → M),
   rel_map (ϕ.on_relation R) x = rel_map R x)
 
-attribute [simp] is_expansion_on.map_on_function is_expansion_on.map_on_relation
+@[simp] lemma map_on_function (M : Type*)
```

Comment:
Are `map_on_function` and `is_expansion_on.map_on_function'` syntactically different?
If it is because you want `M` explicit, it might help to add `[]` after the field names in the class declaration. If you want them in this namespace, you can use `export`
Why do you want `M` explicit?

Reply:
Neither `export` or `alias` seem to let me use dot notation, which is what I'd really most like.